src/backend/commands/sequence.c

```diff
@@ -1401,6 +1401,8 @@ init_params(ParseState *pstate, List *options, bool for_identity,
 											   &max_value,
 											   &min_value);
 
+		if(!newtypid) newtypid = typenameTypeId(pstate,defGetTypeName(as_type));
+
```

Comment:
I think we can avoid engine changes here. Can't we do the same inside the hook?

Reply:
Earlier we were getting a schema does not exist error when the UDT was defined under a schema, which was originating from the typenameTypeId function. So I have made the related changes inside the hook; but we need to call the function again after the hook else newtypid would be uninitialised when the query is called from PG endpoint.

Reply:
this can be handled inside hook, only override newtypid at the end of function, so for every other condition it will retain the original value